src/components/organisms/AdminVenueView/components/Spaces/Spaces.tsx

```diff
@@ -0,0 +1,248 @@
+    template: VenueTemplate.conversationspace,
+  },
+  {
+    text: "Auditorium",
+    icon: RoomIconAuditorium,
+    template: VenueTemplate.audience,
+  },
+  {
+    text: "Music Bar",
+    icon: RoomIconMusicBar,
+    template: VenueTemplate.jazzbar,
+  },
+  {
+    text: "Burn Barrel",
+    icon: RoomIconBurnBarrel,
+    template: VenueTemplate.firebarrel,
+  },
+  {
+    text: "Art Piece",
+    icon: RoomIconArtPiece,
+    template: VenueTemplate.artpiece,
+  },
+  {
+    text: "Experience",
+    icon: RoomIconExperience,
+    template: VenueTemplate.zoomroom,
+  },
+  {
+    text: "External link",
+    icon: RoomIconExternalLink,
+    template: RoomTemplate.external,
+  },
+  {
+    text: "Map",
+    icon: RoomIconMap,
+    template: VenueTemplate.partymap,
+  },
+];
+
+export interface SpacesProps {
+  venue: Venue_v2;
+  onClickNext: () => void;
+}
+
+const emptyRoomsArray: RoomData_v2[] = [];
+
+export const Spaces: React.FC<SpacesProps> = ({ venue, onClickNext }) => {
+  const [selectedRoom, setSelectedRoom] = useState<RoomData_v2>();
+  const [updatedRoom, setUpdatedRoom] = useState<RoomData_v2>({});
+
+  const { isShown: showRooms, toggle: toggleShowRooms } = useShowHide(false);
+  const { isShown: showAddRoom, toggle: toggleShowAddRoom } = useShowHide(
+    false
+  );
+  const {
+    isShown: showAdvancedSettings,
+    toggle: toggleShowAdvancedSettings,
+  } = useShowHide(false);
+
+  const history = useHistory();
+
+  const hasSelectedRoom = !!selectedRoom;
+  const numberOfRooms = venue.rooms?.length ?? 0;
+
+  const clearSelectedRoom = useCallback(() => {
+    setSelectedRoom(undefined);
+    setUpdatedRoom({});
+  }, []);
+
+  const updateRoomPosition = useCallback(async (position: Position) => {
+    if (!position) return;
+
+    setUpdatedRoom((room) => ({
+      ...room,
+      x_percent: position.left,
+      y_percent: position.top,
+    }));
+  }, []);
+
+  const updateRoomSize = useCallback(async (size: Dimensions) => {
+    if (!size) return;
+
+    setUpdatedRoom((room) => ({
+      ...room,
+      width_percent: size.width,
+      height_percent: size.height,
+    }));
+  }, []);
+
+  const renderVenueRooms = useMemo(
+    () =>
+      venue.rooms?.map((room, index) => (
+        <div
+          key={`${index}-${room.title}`}
+          className="Spaces__venue-room"
+          onClick={() => setSelectedRoom(room)}
+        >
+          <div
+            className="Spaces__venue-room-logo"
+            style={{ backgroundImage: `url(${room.image_url})` }}
+          />
+          <div className="Spaces__venue-room-title">{room.title}</div>
+        </div>
+      )),
+    [venue.rooms]
+  );
+
+  const renderAddRooms = useMemo(
+    () =>
+      venueRooms.map((venueRoom, index) => (
+        <VenueRoomItem
+          key={`${venueRoom.text}-${index}`}
+          text={venueRoom.text}
+          template={venueRoom.template}
+          icon={venueRoom.icon}
+        />
+      )),
+    []
```

Comment:
```suggestion
    [venueRooms]
```
?

Reply:
No need, it's a global const which will never change. It doesn't even need readonly. TS is smart enough to know this.